src/hooks/auditoriumSections.ts

```diff
@@ -0,0 +1,134 @@
+import { User } from "types/User";
+
+import { setGridData } from "api/profile";
+
+import {
+  currentAuditoriumSectionsSelector,
+  currentAuditoriumSectionsByIdSelector,
+} from "utils/selectors";
+import { WithId } from "utils/id";
+import { getPositionHash } from "utils/auditorium";
+
+import { useSelector } from "./useSelector";
+import { useFirestoreConnect, isLoaded } from "./useFirestoreConnect";
+import { useRecentVenueUsers } from "./users";
+import { useUser } from "./useUser";
+import { useCallback, useMemo } from "react";
+
+const useConnectAuditoriumSections = (venueId?: string) => {
+  useFirestoreConnect(() => {
+    if (!venueId) return [];
+
+    return [
+      {
+        collection: "venues",
+        doc: venueId,
+        subcollections: [{ collection: "sections" }],
+        storeAs: "currentAuditoriumSections",
+      },
+    ];
+  });
+};
+
+export interface UseAuditoriumSectionProps {
+  sectionId: string;
+  venueId?: string;
+}
+
+export const useAuditoriumSection = ({
+  venueId,
+  sectionId,
+}: UseAuditoriumSectionProps) => {
+  useConnectAuditoriumSections(venueId);
+
+  const { userWithId } = useUser();
+  const userId = userWithId?.id;
+
+  const sections = useSelector(currentAuditoriumSectionsByIdSelector);
+  const section = sections?.[sectionId];
+
+  const seatedUsers = useSectionSeatedUsers(venueId, sectionId);
+
+  const seatedUsersByHash = useMemo(
+    () =>
+      seatedUsers.reduce<Record<string, WithId<User> | undefined>>(
```

Comment:
```suggestion
      seatedUsers.reduce<Partial<Record<string, WithId<User>>>(
```

Reply:
Though I wonder, would it be cleaner for us to use `Map` here rather than a plain object (personally I am thinking yes)?

- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map
  - https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map#objects_vs._maps
    - > there are important differences that make Map preferable in certain cases

Reply:
good idea

Reply:
fixed in https://github.com/sparkletown/sparkle/pull/1251/commits/adf6f1b18663d0e43209d78822eb78a1f85af80d

Reply:
See https://github.com/sparkletown/sparkle/commit/adf6f1b18663d0e43209d78822eb78a1f85af80d#r49201035